### PR TITLE
Ensure we deal with PossiblyNormalCompletions properly

### PR DESCRIPTION
**Pull request by trueadm**
Release notes: none

Added missing logic that handles `PossiblyNormalCompletion` from effects.

### CODE REVIEW

File: src/evaluators/CallExpression.js [hunk 1]
```diff
@@ -87,9 +87,6 @@ function callBothFunctionsAndJoinTheirEffects(
     // all the branches come together into one.
     completion = realm.composeWithSavedCompletion(completion);
   }
-
```
Comment (hermanventer) [inline src/evaluators/CallExpression.js]: Why delete the comment?

File: src/evaluators/UnaryExpression.js [hunk 2]
```diff
@@ -139,8 +139,15 @@ function tryToEvaluateOperationOrLeaveAsAbstract(
...
-  realm.applyEffects(effects);
   let completion = effects[0];
+  if (completion instanceof PossiblyNormalCompletion) {
+    // in this case one of the branches may complete abruptly, which means that
+    // not all control flow branches join into one flow at this point.
+    // Consequently we have to continue tracking changes until the point where
+    // all the branches come together into one.
+    completion = realm.composeWithSavedCompletion(completion);
+  }
+  realm.applyEffects(effects);
```
Comment (hermanventer) [inline src/evaluators/UnaryExpression.js]: Comment?

File: src/evaluators/CallExpression.js (hunk 1 above)
Comment (trueadm) [reply src/evaluators/CallExpression.js]: I didn't delete it, it was still there. I force pushed the previous change. You were too ninja quick and saw before I got a chance to fix the push :D

File: src/evaluators/UnaryExpression.js (hunk 2 above)
Comment (trueadm) [reply src/evaluators/UnaryExpression.js]: Was there, I force pushed the previous commit by mistake.

**Pull request closed without merging**